src/coreclr/src/jit/codegenlinear.cpp

```diff
@@ -2049,6 +2139,25 @@ void CodeGen::genProduceReg(GenTree* tree)
                     }
                 }
             }
+            else if (tree->IsMultiRegLclVar())
+            {
+                assert(compiler->lvaEnregMultiRegVars);
+                GenTreeLclVar* lclNode  = tree->AsLclVar();
+                LclVarDsc*     varDsc   = compiler->lvaGetDesc(lclNode->GetLclNum());
+                unsigned       regCount = varDsc->lvFieldCnt;
+                for (unsigned i = 0; i < regCount; i++)
+                {
+                    if (!lclNode->IsLastUse(i))
```

Comment:
For the non-multi-reg case, where we have a candidate lclVar, we're checking that `GTF_VAR_DEATH` is 0, which is the same as `lclNode->IsLastUse(0)`. Note that a multi-reg lclVar won't return true for `genIsRegCandidateLocal(tree)` since its `gtLclNum` is the parent (which isn't a reg candidate). For the multi-reg lclVar we check each field's last-use bit separately.